app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationActionBarView.kt

```diff
@@ -92,37 +96,58 @@ class ConversationActionBarView @JvmOverloads constructor(
 
     fun updateSubtitle(recipient: Recipient, openGroup: OpenGroup? = null, config: ExpirationConfiguration? = null) {
         val settings = mutableListOf<ConversationSetting>()
+
+        // Specify the disappearing messages subtitle if we should
         if (config?.isEnabled == true) {
-            val prefix = when (config.expiryMode) {
-                is ExpiryMode.AfterRead -> R.string.expiration_type_disappear_after_read
-                else -> R.string.expiration_type_disappear_after_send
-            }.let(context::getString)
+            // Get the type of disappearing message and the abbreviated duration..
+            val dmTypeString = when (config.expiryMode) {
+                is AfterRead -> context.getString(R.string.read)
+                else -> context.getString(R.string.send)
+            }
+            val durationAbbreviated = ExpirationUtil.getExpirationAbbreviatedDisplayValue(config.expiryMode.expirySeconds)
+
+            // ..then substitute into the string..
+            val subtitleTxt = Phrase.from(context, R.string.disappearingMessagesDisappear)
+                .put(DISAPPEARING_MESSAGES_TYPE_KEY, dmTypeString)
+                .put(TIME_KEY, durationAbbreviated)
+                .format().toString()
+
+            // .. and apply to the subtitle.
             settings += ConversationSetting(
-                "$prefix - ${ExpirationUtil.getExpirationAbbreviatedDisplayValue(context, config.expiryMode.expirySeconds)}",
+                subtitleTxt,
                 ConversationSettingType.EXPIRATION,
                 R.drawable.ic_timer,
-                resources.getString(R.string.AccessibilityId_disappearing_messages_type_and_time)
+                resources.getString(R.string.AccessibilityId_disappearingMessagesDisappear)
             )
         }
+
         if (recipient.isMuted) {
             settings += ConversationSetting(
                 recipient.mutedUntil.takeUnless { it == Long.MAX_VALUE }
-                    ?.let { context.getString(R.string.ConversationActivity_muted_until_date, DateUtils.getFormattedDateTime(it, "EEE, MMM d, yyyy HH:mm", Locale.getDefault())) }
-                    ?: context.getString(R.string.ConversationActivity_muted_forever),
+                    ?.let {
+                        val mutedDuration = (it - System.currentTimeMillis()).milliseconds
+                        val durationString = LocalisedTimeUtil.getDurationWithSingleLargestTimeUnit(context, mutedDuration)
+                        Phrase.from(context, R.string.notificationsMuteFor)
```

Comment:
There are so many of these long ugly calls to `Phrase` now... This would have been a good thing to set as an extension function on context...
`context.getString(R.string.notificationsMuteFor, TIME_LARGE_KEY, durationString)` or something like that.

Reply:
`this` is already a context so you won't need that plus the extension can already return the `toString()` as well.
So ideally
```
Phrase.from(context, R.string.notificationsMuteFor)
                            .put(TIME_LARGE_KEY, durationString)
                            .format().toString()
```
would become
`context.getString(R.string.notificationsMuteFor, TIME_LARGE_KEY, durationString)`